appscale/tools/agents/ec2_agent.py

```diff
@@ -200,37 +220,69 @@ def create_security_group(self, parameters, group):
       "name {0}".format(group))
 
 
-  def authorize_security_group(self, parameters, group, from_port, to_port,
-    ip_protocol, cidr_ip):
+  def get_security_group_by_name(self, parameters, group=None):
+    """Gets a security group in AWS with the given name.
+
+    Args:
+      parameters: A dict that contains the credentials necessary to authenticate
+        with AWS.
+      group: A str that names the group that should be found. Or None to
+        use the group in parameters.
+    Returns:
+      The 'boto.ec2.securitygroup.SecurityGroup' that has the correct group
+      name.
+    Raises:
+      AgentRuntimeException: If the security group could not be found.
+    """
+    # Get security group.
+    if not group:
```

Comment:
Given the name of the function, I don't think client should be permitted to omit the group name.